### ISSUE TITLE
Archive old projects

### ISSUE THREAD

**Issue by upstreamplugin** [upstreamplugin/UpStream#236]
> As time progresses, If we have many old projects(over hundred).
> Can we *archive* our all related projects?

This could be useful. I imagine the frontend dashboard will get very busy in this situation

**Comment by tardis221** [comment]
Definitely needed! Please add my vote for this feature.

**Comment by upstreamplugin** [comment]
Maybe anything marked as "Closed" or "Completed" should be hidden by default? That goes for Tasks, Bugs, Projects etc.

Or perhaps this should be configurable in the "Status" settings

**Comment by upstreamplugin** [comment]
This could be just a small tweak to the default settings in the new filters

**Comment by b-trav** [comment]
Hi @andergmartins @stevejburge ,

I thought I should expand a little on my use-case for this. We are a group in a university department who have started using your plugin for our group and for final year engineering students doing a thesis project. In the future, we might have a long list of closed projects. It would be nice if we did not need to delete them, but could instead archive them. I can imagine a list which is so long that pulling them out of the MySQL database, processing them in php and then hiding them in the table using CSS would considerably slow down the page.

How does a site like github deal with this issue? I haven't checked, but I assume that it does not give you all the "closed" issues until you ask for them.

The same principles hold true for tasks (see issue #544 ) as well as projects.

**Comment by andergmartins** [comment]
@stevejburge yes, but the problem is how the filters work on UpStream (at least for tasks, bugs, etc.). Unfortunately, they do not load the data using Ajax. But it loads all the items and filter on the runtime, using JS only. So we would need to remove the closed statuses from the filters as well until we fix all this.

We need to improve the way the tasks/bugs are stored in the project, so we can query only the ones we want, not all. Right now, they are stored as a block of text, serialized. We need to convert them in a specific post type, so we can make it load only what we want. 

Otherwise, we will be just hidden in the interface, but not from the page processing.

**Comment by andergmartins** [comment]
I have implemented 2 options: Filter closed items/projects, where the closed items will be loaded, but the filter is prefilled with all "open" statuses. The second option allows to archive closed projects/items, so they don't load in the frontend.
